### PR TITLE
test: debounce hook test

### PR DESCRIPTION
**Pull request by arjita-mitra**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - Unit test for debounce hook

- The **PR Description**
  - Unit test for debounce hook
----

### CODE REVIEW

File: src/script/hooks/useDebounce.test.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+  it('should not call the callback if it is unmounted before the debounce time', () => {
+    const callback = jest.fn();
+    const time = 1000;
+
+    const {unmount} = renderHook(() => useDebounce(callback, time));
+
+    act(() => {
+      unmount();
+      jest.advanceTimersByTime(500);
```
Comment (PatrykBuniX) [inline src/script/hooks/useDebounce.test.tsx]: Shouldn't this time be more than 1000 here? To make sure callback would run if component wasn't unmounted? This tests will pass even when your remove `unmount()` from here.

Comment (arjita-mitra) [reply src/script/hooks/useDebounce.test.tsx:72]: I got your point but this test is - should not call the callback if it is unmounted before the debounce time

we need to make it - should not call the callback if it is unmounted after the debounce time

Comment (arjita-mitra) [reply src/script/hooks/useDebounce.test.tsx]: actually before/after time doesn't matter. if the component is unmounted before you advanced the timer to something > 1000 / <1000 the callback won't be called. so the important thing here is the component is unmounted before the timer could finish

Comment (PatrykBuniX) [reply src/script/hooks/useDebounce.test.tsx]: `should not call the callback if it is unmounted before the debounce time` callback will never be called in this case, doesn't matter if we unmount or not. So this is exactly the same thing you're testing in the first case - that callback should not run before debounce time passes

Comment (arjita-mitra) [reply src/script/hooks/useDebounce.test.tsx]: I think it makes more sense to advance the timer above 1000 to make the purpose of the test more clear

**Review by PatrykBuniX — APPROVED** [review]
Nice!

**Pull request merged**